lib/HttpClient.ts

```diff
@@ -257,8 +260,52 @@ export class HttpClient {
 
         info.parsedUrl = url.parse(requestUrl);
         let usingSsl = info.parsedUrl.protocol === 'https:';
-        info.httpModule = usingSsl ? https : http;
         var defaultPort: number = usingSsl ? 443 : 80;
+        info.httpModule = usingSsl ? https : http;
+        info.options = <http.RequestOptions>{};
+        info.options.host = info.parsedUrl.hostname;
+        info.options.port = info.parsedUrl.port ? parseInt(info.parsedUrl.port) : defaultPort;
+        info.options.path = (info.parsedUrl.pathname || '') + (info.parsedUrl.search || '');
+        info.options.method = method;
+        info.options.headers = headers || {};
+        info.options.headers["User-Agent"] = this.userAgent;
+        
+        let agent;
+        if (!this._agent) {
+            agent = this._getRequestAgent(requestUrl);
+            if (this._reuseAgent) {
+                this._agent = agent;
+            }
+        }
+        else {
+            agent = this._agent
+        }
+
+        info.options.agent = agent;
+
+        // gives handlers an opportunity to participate
+        if (this.handlers) {
+            this.handlers.forEach((handler) => {
+                handler.prepareRequest(info.options);
+            });
+        }
+
+        return info;
+    }
+
+    private _getRequestAgent(requestUrl) {
+        let agent;
+        let keepAlive = false;
+        let maxSockets = http.globalAgent.maxSockets;
+
+        if (!!this.requestOptions) {
+            keepAlive = !!this.requestOptions.keepAlive;
```

Comment:
add destroy agent support to make sure connections get closed after use

Reply:
done